src/objectExplorer/objectExplorerService.ts

```diff
@@ -484,10 +492,17 @@ export class ObjectExplorerService {
             if (!(<IConnectionProfile>node.connectionCredentials).savePassword) {
                 node.connectionCredentials.password = '';
             }
+            let disconnectedNode;
             // make a new node to show disconnected behavior
-            let disconnectedNode = new TreeNodeInfo(node.label as string, Constants.disconnectedServerLabel,
-                node.collapsibleState, node.nodePath, node.nodeStatus, Constants.disconnectedServerLabel,
-                undefined, node.connectionCredentials, node.parentNode);
+            if (typeof node.label === 'string') {
```

Comment:
same here:
`
const label = typeof node.label === 'string' ? node.label:node.label.label;
`

Reply:
giving the fact that it is being used so many times, you can create a util function to get the label from a node.

Reply:
ok that makes a lot of sense, I knew there had to be a simpler way